swap_meet/vendor.py

```diff
@@ -1,2 +1,71 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        inventory_by_cat_list = [
+            item for item in self.inventory if item.category == category]
+
+        return inventory_by_cat_list
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory or their_item not in friend.inventory:
+            return False
+
+        self.add(their_item)
+        self.remove(my_item)
+        friend.add(my_item)
+        friend.remove(their_item)
+
+        return True
+
+    def swap_first_item(self, friend):
+        if not self.inventory or not friend.inventory:
+            return False
+
+        self.swap_items(friend, self.inventory[0], friend.inventory[0])
+
+        return True
```

Comment:
Great use of helper functions! 

Currently, swapping items has a time complexity of O(n) due to the `remove` method in `swap_items`. We can make a subtle improvement to O(1) by using multiple assignment instead:

```python
self.inventory[0], other.inventory[0] = other.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp